File: src/main/kotlin/dev/eca/ecaintellij/HelloAction.kt
package dev.eca.ecaintellij


import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ui.Messages
import com.intellij.openapi.diagnostic.Logger

class HelloAction : AnAction() {

    private val logger = Logger.getInstance(HelloAction::class.java)

    override fun actionPerformed(e: AnActionEvent) {
        logger.info("Hello from ECA!")

        Messages.showInfoMessage(
            "Hello from ECA!",
            "ECA Plugin"
        )
    }
}

File: src/main/clojure/dev/eca/ecaintellij/hello_action.clj
(ns dev.eca.ecaintellij.hello-action
  (:require [com.github.ericdallo.clj4intellij.action :as action])
  (:import
     [com.intellij.openapi.actionSystem CommonDataKeys]
     [com.intellij.openapi.actionSystem AnActionEvent]
     [com.intellij.openapi.ui Messages]
     [com.intellij.openapi.editor Editor]))

(defn on-performed [^AnActionEvent event]
  (.showInfoMessage Messages "Hello from ECA with clojure!" "ECA Plugin"))

(defn hello-action
  []
  (action/register-action!
    :on-performed on-performed))